src/project_model.rs

```diff
@@ -40,12 +40,12 @@ pub struct Dep {
 }
 
 impl ProjectModel {
-    pub fn load(manifest: &Path, vfs: &Vfs) -> Result<ProjectModel, failure::Error> {
-        assert!(manifest.ends_with("Cargo.toml"));
+    pub fn load(ws_manifest: &Path, vfs: &Vfs) -> Result<ProjectModel, failure::Error> {
+        assert!(ws_manifest.ends_with("Cargo.toml"));
         let mut config = Config::default()?;
-        // frozen=true, locked=true
-        config.configure(0, Some(true), &None, true, true, &None, &[])?;
-        let ws = Workspace::new(&manifest, &config)?;
+        // frozen = false, locked = false
```

Comment:
It's definitely not making things any more worse/more complicated, since we currently [https://github.com/rust-lang-nursery/rls/blob/master/src/build/cargo.rs#L572](construct) a Cargo config with defaults (frozen, locked = false) as well.